modules/reactor/doc/content/source/meshgenerators/PinMeshGenerator.md

```diff
@@ -0,0 +1,50 @@
+
+!alert! note title=Pin block IDs are modified to match region IDs
+It should be noted here that both the extra integer "region_id" and the block ID of the resultant pin elements will be modified to match the same value as specified by [!param](/Mesh/PinMeshGenerator/region_ids).
+!alert-end!
+
+The region_ids parameter entries can conveniently be selected to match material ids to be assigned to each region of the problem. Using the same value in multiple entries of the region_id parameter will effectively assign elements in multiple zones to the same subdomain_id and the same region_id. For meshes with all quadrilateral elements, this approach does not present any conflicts. However, if [!param](/Mesh/PinMeshGenerator/quad_center_elements) is set to false, the innermost radial zone of the pin is discretized into triangular elements, and therefore only one radial meshing interval may be defined for this radial zone using the [!param](/Mesh/PinMeshGenerator/mesh_intervals) parameter.  This ensures that a single region id does not correspond to a mesh discretization with both triangle and quadrilateral mesh elements, which will occur if more than one meshing interval is applied to the centermost pin zone with triangular elements. Moreover, the subdomain ID associated with any zone of triangular elements should not be shared with another zone containing quadrilateral elements, otherwise MOOSE will error out.
+
+## Reporting ID Information
+
+The `PinMeshGenerator` object also tags the mesh elements with the extra integer reporting ID named "region_id".
+
+The `PinMeshGenerator` object also automatically tags the mesh with the [!param](/Mesh/PinMeshGenerator/pin_type) using the extra integer name "pin_type_id" and, if extruded, the axial layers using the extra integer name "plane_id".
+
+## Exterior Boundary ID Information
+
+The `PinMeshGenerator` object automatically assigns boundary information derived from the [!param](/Mesh/PinMeshGenerator/pin_type) parameter. The exterior pin boundary is assigned the ID equal to 20000 + the pin type ID and is named "outer_pin_<pin_type_id>" (for example a pin with a pin type ID of 1 will have a boundary ID of 20001 and boundary name of "outer_pin_1").
+
+If the pin is extruded to three dimensions the top-most boundary ID must be assigned using [!param](/Mesh/ReactorMeshParams/top_boundary_id) and will have the name "top", while the bottom-most boundary must be assigned using [!param](/Mesh/ReactorMeshParams/bottom_boundary_id) and will have the name "bottom".
+
+## Example Syntax
+
+!listing modules/reactor/test/tests/meshgenerators/pin_mesh_generator/pin_only.i block=Mesh
+
+!media reactor/meshgenerators/pin_mesh_generator.png
```

Comment:
this is kind of the most atypical pin
Id put the nicest one, round & with clad right etc if I were you